pyasn1/codec/ber/decoder.py

```diff
@@ -4,45 +4,117 @@
 # Copyright (c) 2005-2019, Ilya Etingof <etingof@gmail.com>
 # License: http://snmplabs.com/pyasn1/license.html
 #
+import os
+import sys
+from io import BytesIO, BufferedReader
+
 from pyasn1 import debug
 from pyasn1 import error
 from pyasn1.codec.ber import eoo
 from pyasn1.compat.integer import from_bytes
 from pyasn1.compat.octets import oct2int, octs2ints, ints2octs, null
+from pyasn1.error import PyAsn1Error
 from pyasn1.type import base
 from pyasn1.type import char
 from pyasn1.type import tag
 from pyasn1.type import tagmap
 from pyasn1.type import univ
 from pyasn1.type import useful
 
-__all__ = ['decode']
+
+__all__ = ['decodeStream']
 
 LOG = debug.registerLoggee(__name__, flags=debug.DEBUG_DECODER)
 
 noValue = base.noValue
 
 
+_BUFFER_SIZE = 1024
+_PY2 = sys.version_info < (3,)
+
+
+def asSeekableStream(substrate):
```

Comment:
Is `seekable` important to have in the name here? Looking at Python's `io` class hierarchy, it appears that `seek` is a property of every stream...

Reply:
Depends on your taste but the `seek` method does not mean a stream is `seekable`, despite the API :-(